### PR TITLE
Makefile: clean before do check/test

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

**What type of PR is this?**

/kind cleanup

### CODE REVIEW

File: build/root/Makefile [hunk 1]
```diff
@@ -181,7 +181,7 @@ ifeq ($(PRINT_HELP),y)
 check test:
 	@echo "$$CHECK_TEST_HELP_INFO"
 else
-check test: generated_files
+check test: clean generated_files
```
Comment (neolit123) [inline build/root/Makefile:184]: hi, making "clean" a dependency seems odd in a Makefile rule.
why is this needed?

Comment (daixiang0) [reply build/root/Makefile:184]: The things after test may lead next test failed or use cache, which are not what we want to see.

**Comment by neolit123** [comment]
> daixiang0 requested a review from neolit123 5 hours ago

i'd appreciate if someone else takes a look here.
thanks.

/uncc

**Comment by spiffxp** [comment]
/close
Running `make test` without rebuilding everything from scratch is a common use case, we're not going to break that.  If you can be more descriptive about the actual concrete problems you are running into, there may be a more specific appropriate fix.  But we're not going to accept making `clean` a required target for `test`
